core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStore.java

```diff
@@ -65,21 +66,24 @@ public LocalPageStore(LocalPageStoreOptions options) {
   }
 
   @Override
-  public void put(PageId pageId, byte[] page) throws IOException {
+  public void put(PageId pageId, byte[] page) throws ResourceExhaustedException, IOException {
     Path p = getFilePath(pageId);
-    if (!Files.exists(p)) {
-      Path parent = Preconditions.checkNotNull(p.getParent(),
-          "parent of cache file should not be null");
-      Files.createDirectories(parent);
-      Files.createFile(p);
-    }
     try {
+      if (!Files.exists(p)) {
+        Path parent = Preconditions.checkNotNull(p.getParent(),
+            "parent of cache file should not be null");
+        Files.createDirectories(parent);
+        Files.createFile(p);
+      }
       // extra try to ensure output stream is closed
       try (FileOutputStream fos = new FileOutputStream(p.toFile(), false)) {
         fos.write(page);
       }
     } catch (Exception e) {
       Files.deleteIfExists(p);
+      if (e.getMessage().contains("No space left on device")) {
+        throw new ResourceExhaustedException("No space left on device");
```

Comment:
Make the string a constant? Also is it guaranteed or is there a better way of referencing it?

Reply:
done.
it is not guaranteed, this is best effort for monitoring and recover from this error